src/zh/happymh/src/eu/kanade/tachiyomi/extension/zh/happymh/Happymh.kt

```diff
@@ -119,22 +119,46 @@ class Happymh : HttpSource(), ConfigurableSource {
     // Search
 
     override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
-        val body = FormBody.Builder()
-            .addEncoded("searchkey", query)
-            .add("v", "v2.13")
-            .build()
+        if (query.isNotEmpty()) {
+            val body = FormBody.Builder()
+                .addEncoded("searchkey", query)
+                .add("v", "v2.13")
+                .build()
 
-        val header = headersBuilder()
-            .add("referer", "$baseUrl/sssearch")
-            .build()
+            val header = headersBuilder()
+                .add("referer", "$baseUrl/sssearch")
+                .build()
 
-        return POST("$baseUrl/v2.0/apis/manga/ssearch", header, body)
+            return POST("$baseUrl/v2.0/apis/manga/ssearch", header, body)
+        }
+        val url = "$baseUrl/apis/c/index".toHttpUrl().newBuilder()
+        filters.filterIsInstance<UriPartFilter>().forEach {
+            if (it.selected.isNotEmpty()) {
+                url.addQueryParameter(it.key, it.selected)
+            }
+        }
+        val header = headersBuilder().add("referer", "$baseUrl/latest/${url.build().query}").build()
+        url.addQueryParameter("pn", page.toString())
+        return GET(url.build(), header)
     }
 
     override fun searchMangaParse(response: Response): MangasPage {
+        if (response.request.url.toString().contains("/apis/c/index")) {
```

Comment:
Check against `response.request.url.pathSegments instead.

Reply:
pathSegments returns a List, and to implement this functionality, I'd either need to reassemble it into a String or directly compare the list's values at corresponding positions, which seems to make the implementation more complicate. Please let me know if there's a simpler way or an existing extension method to achieve this quickly.

Reply:
Fair. Can you compare against `encodedPath` instead?
```suggestion
        if (response.request.url.encodedPath == "/apis/c/index") {
```

I have not tested it.